### ISSUE TITLE
Form showing improper error msg in field of type field component

### ISSUE THREAD

**Issue by rajsek** [erikras/redux-form#2531]
Am using Redux 6.0.5. I have a render field component for 'file' input type and i have an onChange event in that file component which gets the uploaded file name and I display it in later.

I have facing problem in error validation function, steps to reproduce:

1. Click the upload button the SYNC validation is triggered(Since there is no file at this moment, which set's 'mandatory error msg')
2. Then when uploading the file, onChange event, I get the uploaded file name and save it in my state
3. Mandatory error message is getting displayed since SYNC validate is occurred before selecting the file

code

```javascript

const labels = defineMessages({
      mandatoryField: {
        id: 'bo_setting_mandatory_field',
        defaultMessage: `This is Mandatory Field`
    }
});
const validate = values => {
    const errors = {};
    const coverPicture = values.get('coverPicture');
    if (!coverPicture)
        errors.coverPicture = labels.mandatoryField;
     return errors;
}

const RenderFileUploadField = ({ input, placeholder, label, buttonLabel, type, meta: { touched, error }, onChange, accept}) => {
    const isError = touched && error;
    return <div className={isError ? 'form-group has-error' : 'form-group'}>
        <label className="control-label">{label}</label>
        <div className="inputGroup">
            <input className="form-control" {...input} type={type} accept={accept} onChange={onChange} />
            <div className="inputFile">
                <small></small>
                <input readOnly={true} name={`${input.name}Text`} type="text" value={placeholder} className="form-control" />
                <span className={isError ? 'error active text-danger' : 'error'}>
                    {isError ? <FormattedMessage id={error.id} defaultMessage={error.defaultMessage} className={isError ? 'error active' : 'error'} />
                        : null}
                </span>
                <span className="inputAddon"><span>{buttonLabel}</span></span>
            </div>
        </div>
    </div>
}

class SettingsForm extends Component {
    handleSubmit(data) {
        console.log("SUBMITTED")
        console.log(data.toJS())
    }
    handleFile(fieldName, event) {
        const { fields } = this.props;
        const files = [...event.target.files];
        if (files.length > 0) {
            this.props.uploadFile(fieldName, files[0].name)
        }
        else if (files.length === 0) {
            this.props.uploadFile(fieldName, '')
        }
        event.preventDefault();
    }
    render() {
        const {formatMessage} = this.props.intl;
        const { handleSubmit, pristine, submitting, error, reset } = this.props;

        return (
                               <form className="equalColumn">
 <Field name="coverPicture" type="file" onChange={this.handleFile.bind(this, 'coverPicture')}
                                        label={formatMessage(labels.defaultCoverPic)}
                                        component={RenderFileUploadField}
                                        buttonLabel={formatMessage(labels.uploadFile)}
                                        placeholder={coverPicture ? coverPicture : formatMessage(labels.defaultCoverPic)}
                                        accept='image/*'
                                        value={null} />
                                        <a href="#" className="btn btn-success" onClick={handleSubmit(this.handleSubmit.bind(this))} >{formatMessage(labels.saveTheSettingsForm)}</a>
                    </form>


        );
    }

}

SettingsForm.propTypes = {
    intl: intlShape.isRequired
};


function mapStateToProps(state) {
    return {
        }
}

const injectIntlDecorator = _.curryRight(injectIntl)
export default compose(injectIntlDecorator(),
    reduxForm({
        form: 'settings',
        destroyOnUnmount: false, 
        validate
    }),
    connect(
        mapStateToProps,
        actions,
        null,
        { pure: false }
    )
)(SettingsForm);

```

In short: It seems like, onClick of Upload button 'OnBlur enter image description here' validation is trigger before selecting file and validation is not triggered after the file is changed... can anyone help me on this?
![image](https://cloud.githubusercontent.com/assets/5106537/22497823/2cd7a25e-e87a-11e6-9bae-dc519e5f8f97.png)

**Comment by gustavohenke** [comment]
Amazingly delayed reply to this issue.
I'm pretty sure it's irrelevant nowadays, please reopen with an up-to-date version if needed.
